server/game/cards/locations/01/castleblack.js

```diff
@@ -1,15 +1,16 @@
 const DrawCard = require('../../../drawcard.js');
 
 class CastleBlack extends DrawCard {
-    setupCardAbilities() {
+    setupCardAbilities(ability) {
         this.action({
             title: 'Kneel this card to stand and give a defending character +2 STR',
+            cost: ability.costs.kneelSelf(),
             method: 'kneel'
         });
     }
 
     kneel(player) {
-        if(this.location !== 'play area' || !this.game.currentChallenge || this.kneeled) {
+        if(this.location !== 'play area' || !this.game.currentChallenge) {
```

Comment:
When using the cost framework, these types of `if`s should be converted to the `condition` property on action instead. Otherwise, the cost will be paid before it discovers that the ability can't be used. https://github.com/cryogen/throneteki/blob/master/docs/implementing-cards.md#checking-ability-restrictions

So in this case:
```javascript
this.action({
    title: 'Kneel this card to stand and give a defending character +2 STR',
    condition: () => this.game.currentChallenge,
    cost: ability.costs.kneelSelf(),
    method: 'kneel'
});
```

(See Wildling Horde, it's similar)